lredis/commands.lua

```diff
@@ -1,79 +1,150 @@
-local methods = {}
+local base_methods = {}
 
-function methods:call(...)
+-- client methods and metatable
+local client_methods = setmetatable({}, { __index = base_methods })
+local client_mt = { __index = client_methods }
+local new_client
+
+-- transaction methods and metatable
+local transaction_methods = setmetatable({}, { __index = base_methods })
+local transaction_mt = { __index = transaction_methods }
+local new_transaction
+
+--[[
+--	Base methods
+--]]
+
+function base_methods:call_no_err(...)
 	local resp = self:pcall(...)
-	local is_table = type(resp) == "table"
-	if is_table and resp.err then
-		error(resp.err, 2)
+	if type(resp) ~= "table" or (not resp.ok and not resp.err) then
+		resp = { data = resp }
 	end
 	return resp
 end
 
-local function handle_ok_or_err(resp, lvl)
-	local is_table = type(resp) == "table"
-	if is_table and resp.ok then
-		return resp.ok
-	else
-		local err
-		if is_table and resp.err then
-			err = resp.err
-		else
-			err = "unexpected response format"
-		end
-		if lvl == nil then
-			lvl = 2
-		elseif lvl ~= 0 then
-			lvl = lvl + 1
-		end
-		error(err, lvl)
+function base_methods:call(...)
```

Comment:
There was intentionally no methods on objects that didn't map to redis commands. Otherwise if redis added a "CALL" command then we'd have a conflict

Reply:
Sorry, I don't understand your argument.  Look at your existing `cqueues.lua`:
```lua
function methods:pcallt(...)
function methods:pcall(...)
function methods:start_subscription_modet(...)
function methods:start_subscription_mode(...)
function methods:get_next(...)
```

That directly contradicts your statement.  Those are not native Redis commands, what happens if they are added in the future?